python/tests/test_topology.py

```diff
@@ -49,6 +49,532 @@ def generate_segments(n, sequence_length=100, seed=None):
     return segs
 
 
+def kc_distance_tree(tree1, tree2, topo_v_age=0):
+    """
+    Returns the Kendall-Colijn distance between the specified
+    pair of trees. topology_vs_age determines weight of topology vs
+    branch lengths in calculating the distance.
+    """
+    samples = tree1.tree_sequence.samples()
+    if not np.array_equal(samples, tree2.tree_sequence.samples()):
+        raise ValueError("Trees must have the same samples")
+    if not len(tree1.roots) == len(tree2.roots) == 1:
+        raise ValueError("Trees must have one root")
+    k = samples.shape[0]
+    n = (k * (k - 1)) // 2
+    m = [np.ones(n + k), np.ones(n + k)]
+    M = [np.zeros(n + k), np.zeros(n + k)]
+    for tree_index, tree in enumerate([tree1, tree2]):
+        stack = [(tree.root, 0, tree.time(tree.root))]
+        while len(stack) > 0:
+            u, depth, time = stack.pop()
+            children = tree.children(u)
+            for v in children:
+                stack.append((v, depth + 1, tree.time(v)))
+            for c1, c2 in itertools.combinations(children, 2):
+                for u in tree.samples(c1):
+                    for v in tree.samples(c2):
+                        if u < v:
+                            a, b = u, v
+                        else:
+                            a, b = v, u
+                        pair_index = a * (a - 2 * k + 1) // -2 + b - a - 1
+                        assert m[tree_index][pair_index] == 1
+                        m[tree_index][pair_index] = depth
+                        M[tree_index][pair_index] = tree.time(tree.root) - time
+            if len(tree.children(u)) == 0:
+                M[tree_index][u + n] = tree.branch_length(u)
+    return ((1 - topo_v_age) * np.linalg.norm(m[0] - m[1])) + (
+        topo_v_age * np.linalg.norm(M[0] - M[1]))
```

Comment:
Are you sure this is equivalent to
```
np.linalg.norm((1 - topo_v_age) * (m[0] - m[1]) + topo_v_age * (M[0] - M[1]))
```
? Because that's what I understood the metric to be. From thinking about it quickly, I don't think the two expressions are the same, because you can't distribute `np.linalg.norm` aka Euclidean distance over a `+` operation.

Reply:
Just looked into this, you're right: they're generally not the same thing. They're only the same when ||A+B|| is orthogonal, and that only happens when the topologies between the two trees your comparing is the same. In practice the difference between the distributed version and the correct version is small in absolute terms, but I'll amend it to be correct. Great catch! Thanks Brian